presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java

```diff
@@ -33,11 +33,9 @@ private AgentServer() {}
     public static void main(String[] args)
             throws Exception
     {
-        // TODO: Replace placeholder properties with proper configuration
-        System.setProperty("node.environment", "test");
-        System.setProperty("http-server.http.port", "8081");
-        System.setProperty("log.levels-file", "etc/log.properties");
-        System.setProperty("discovery.uri", "http://localhost:8088");
+        if (System.getProperty("config", null) == null) {
+            System.setProperty("config", "etc/agent.properties");
```

Comment:
Yes, the configuration has to be present for each Agent. It would have to be included when the Agent is installed. There is no way around this, because the Agent can't be remotely configured without already having access to the discovery URI and server port.

This sets the default configuration for the Agent, and uses `etc/agent.properties` if the `config` system property is not set.